build-system/tasks/video-interface-list.js

```diff
@@ -0,0 +1,127 @@
+const fastGlob = require('fast-glob');
+const tempy = require('tempy');
+const {basename} = require('path');
+const {getStdout} = require('../common/process');
+const {green, red} = require('kleur/colors');
+const {log, logWithoutTimestamp} = require('../common/logging');
+const {readFileSync, writeFileSync} = require('fs-extra');
+
+const TASK = 'video-interface-list';
+
+const filepath = 'spec/amp-video-interface.md';
+
+const excludeGeneric = ['amp-video', 'amp-video-iframe'];
+
+const grepJsContent = '"@implements {.*VideoInterface}"';
+
+const grepJsFiles = 'extensions/**/*.js';
+
+// File should have a section containing only entries.
+// - [amp-whatever](https://amp.dev/documentation/components/amp-whatever)
+const entry = (name) =>
+  `-   [${name}](https://amp.dev/documentation/components/${name})`;
+
+const findVideoInterfaceExtensions = () =>
+  getStdout(
+    ['grep -lr', grepJsContent, ...fastGlob.sync(grepJsFiles)].join(' ')
+  )
+    .trim()
+    .split('\n')
+    .map((path) => path.substr('extensions/'.length).split('/').shift())
+    .filter((name) => !excludeGeneric.includes(name));
+
+function getSectionRange(content) {
+  const entryRegExp = new RegExp(
+    entry('$NAME')
+      .replace(/[\[\]\(\)]/g, (c) => `\\${c}`)
+      .replace(/\$NAME/g, '([a-z0-9-]+)') + '[\\n\\s]+',
+    'gim'
+  );
+
+  let match;
+  let sectionStart = -1;
+  let sectionEnd = 0;
+  while ((match = entryRegExp.exec(content)) !== null) {
+    const [line, name] = match;
+    const {index} = match;
+    if (excludeGeneric.includes(name)) {
+      continue;
+    }
+    if (sectionStart < 0) {
+      sectionStart = index;
+      sectionEnd = index;
+    }
+    if (sectionEnd < index) {
+      break;
+    }
+    sectionEnd += line.length;
+  }
+  return [sectionStart, sectionEnd];
+}
+
+function expected(content) {
+  const [sectionStart, sectionEnd] = getSectionRange(content);
+  return (
+    content.substr(0, sectionStart) +
+    findVideoInterfaceExtensions().map(entry).join('\n') +
+    '\n\n' +
+    content.substr(sectionEnd)
+  );
+}
+
+async function videoInterfaceList() {
+  const content = readFileSync(filepath).toString();
+  const output = expected(content);
+
+  if (output === content) {
+    return;
+  }
+
+  const temporary = tempy.file({name: basename(filepath)});
+
+  writeFileSync(temporary, output);
+
+  const diff = getStdout(`diff --unified ${filepath} ${temporary}`)
+    // we don't want to output the ugly temporary name
+    .replace(temporary, filepath);
+
+  log(
+    argv.write
+      ? green(`Wrote ${filepath}:`)
+      : red(`${filepath} requires changes:`)
+  );
+
+  logWithoutTimestamp(diff);
+
+  if (!argv.write) {
+    throw new Error(`You should update this file:\n\tgulp ${TASK} --write\n`);
+  }
+
+  writeFileSync(filepath, output);
+}
+
+module.exports = {
+  [TASK]: videoInterfaceList,
+};
+
+videoInterfaceList.description = `Check or update list on ${filepath}`;
```

Comment:
Maybe state this more simply like `Check or update documentation for 3p video interfaces`?